source/sundials/kinsol.cc

```diff
@@ -354,17 +373,26 @@ namespace SUNDIALS
 
     kinsol_mem = KINCreate();
 
-    int status =
-      KINInit(kinsol_mem, residual_or_iteration_callback<VectorType>, solution);
+    int status = 0;
     (void)status;
-    AssertKINSOL(status);
 
     status = KINSetUserData(kinsol_mem, static_cast<void *>(this));
     AssertKINSOL(status);
 
+    // This must be called before KINSetMAA
     status = KINSetNumMaxIters(kinsol_mem, data.maximum_non_linear_iterations);
     AssertKINSOL(status);
 
+    // From the manual: this must be called BEFORE KINInit
```

Comment:
how bizarre...

Reply:
@bangerth this was the bug. @sebproell discovered that if this function is called after the initialisation a vector of N_Vector is left empty so the fixed point iterations end up segfaulting. Then we looked carefully at the manual and discovered the fix